k8s/core/events.go

```diff
@@ -28,3 +33,33 @@ func (c *Client) ListEvents(namespace string, opts metav1.ListOptions) (*corev1.
 	}
 	return c.kubernetes.CoreV1().Events(namespace).List(opts)
 }
+
+// RecorderOps is an interface to record k8s events
+type RecorderOps interface {
+	// RecordEvent records an event into k8s using client-go's EventRecorder inteface
+	// It takes the event source and the object on which the event is being raised.
+	RecordEvent(source v1.EventSource, object runtime.Object, eventtype, reason, message string)
+}
+
+func (c *Client) RecordEvent(source v1.EventSource, object runtime.Object, eventtype, reason, message string) {
+	c.eventRecordersLock.Lock()
+	if len(c.eventRecorders) == 0 {
+		c.eventRecorders = make(map[string]record.EventRecorder)
+		c.eventBroadcaster = record.NewBroadcaster()
+		c.eventBroadcaster.StartRecordingToSink(
+			&typedcorev1.EventSinkImpl{
+				Interface: c.kubernetes.CoreV1().Events(""), // use the namespace from the object
+			},
+		)
+	}
+	eventRecorder, exists := c.eventRecorders[source.Component]
+	if !exists {
+		eventRecorder = c.eventBroadcaster.NewRecorder(
+			scheme.Scheme,
+			source,
+		)
+		c.eventRecorders[source.Component] = eventRecorder
```

Comment:
Should we use the key as combination of `source.Component + source.Host`? 
Not sure if we need it, but in case two sources have same component name but different host name, events might get created with the first source that comes here. 
Example I was thinking was, let's say we don't know the k8s node name and use node id but later want to use the k8s node name or the other way round.

Reply:
sounds good. Will change it to "source.Component-source.Host"